TM1py/Services/HierarchyService.py

```diff
@@ -634,8 +647,13 @@ def update_or_create_hierarchy_from_dataframe(
                 sum_numeric_duplicates=False,
                 use_blob=True)
 
-        if unwind:
+        if unwind_all:
             self.remove_all_edges(dimension_name, hierarchy_name)
+        else:
+            if unwind_consolidation is not None:
+                for elem in unwind_consolidation:
+                    if self.elements.exists(dimension_name, hierarchy_name, elem):
+                        self.remove_edges_under_consolidation(dimension_name, hierarchy_name, elem)
```

Comment:
The `remove_edges_under_consolidation` function is fine, but it's not the most efficient.

That's because it retrieves the entire hierarchy, deletes some edges on the py object, and then sends it back to TM1. 

It's more efficient if you find all descendants with MDX and use the function in the `ElementService` to remove edges.


``` python
    edges= tm1.elements.get_edges_under_consolidation(
        dimension_name="d1",
        hierarchy_name="d1",
        consolidation="c1"
    )
    tm1.elements.delete_edges_use_ti(
        dimension_name="d1", 
        hierarchy_name="d1", 
        edges=edges)
```

Reply:
@MariusWirtz  thanks for your suggestion with this. I have taken a mixed approach here,

1. Creating an unique dictionary of all edges for all consolidations in the list using `get_edges_under_consolidation()`. Added a function ` join()` in Utils.py for `CaseAndSpaceInsensitiveTuplesDict` class to merge the resulting edges for each consol.

2. Using the master edges dictionary, calling `delete_edges()` once to unwind all edges. I am using _use_ti_ based on admin validation as param, once my other pull request is approved. I can switch it to _use_blob_ option (I think we need to update this flag in various places).

### Reference Code:
### HierarchyService.py

```
        if unwind_all:
            self.remove_all_edges(dimension_name=dimension_name, hierarchy_name=hierarchy_name)
        else:
            if unwind_consolidation is not None:
                edges = CaseAndSpaceInsensitiveTuplesDict()
                for elem in unwind_consolidation:
                    if self.elements.exists(dimension_name=dimension_name, hierarchy_name=hierarchy_name, element_name=elem):
                        temp_edges = self.elements.get_edges_under_consolidation(
                                            dimension_name=dimension_name, 
                                            hierarchy_name=hierarchy_name, 
                                            consolidation=elem)
                        edges.join(temp_edges)
                self.elements.delete_edges(
                            dimension_name=dimension_name, 
                            hierarchy_name=hierarchy_name,
                            edges=edges,
                            use_ti=self.is_admin)  

```

###  Utils.py
```

    # Join two dictionaries together
    def join(self, other):
        if isinstance(other, CaseAndSpaceInsensitiveTuplesDict):
            for key,value in other.items():
                self._store[tuple([lower_and_drop_spaces(item) for item in key])] = (key, value)
        else:
             raise ValueError("Other object is not of type CaseAndSpaceInsensitiveTuplesDict")
```